idaes/core/base/tests/test_flowsheet_model.py

```diff
@@ -400,6 +400,39 @@ def test_dynamic_external_time(self):
         assert m.fs.sub.config.time is m.s
         assert m.fs.sub.time_units == units.minute
 
+    @pytest.mark.unit
+    def test_dynamic_nested_time(self):
+        m = ConcreteModel()
+        m.s1 = ContinuousSet(initialize=[4, 5])
+        m.s2 = ContinuousSet(initialize=[1,2,3])
+        m.fs = FlowsheetBlock(dynamic=True, time_units=units.s)
+        m.fs.sub = FlowsheetBlock(dynamic=True, time=m.s2, time_units=units.s)
+
+        assert m.fs.sub.config.dynamic is True
+        assert isinstance(m.fs.sub.time, ContinuousSet)
+        assert m.fs.sub.time_units == units.s
+
+    @pytest.mark.unit
+    def test_dynamic_nested_time_from_time_set(self):
+        m = ConcreteModel()
+        m.fs = FlowsheetBlock(dynamic=True, time_set = [1,2,3], time_units=units.s)
+        m.fs.sub = FlowsheetBlock(dynamic=True, time_set = [1, 2], time_units=units.s)
+
+        assert m.fs.sub.config.dynamic is True
+        assert isinstance(m.fs.sub.time, ContinuousSet)
+        assert m.fs.sub.time_units is units.s
+
+    @pytest.mark.unit
+    def test_dynamic_parent_time_indexed_ss_child(self):
+        m = ConcreteModel()
+        m.fs = FlowsheetBlock(dynamic=True, time_set = [1,2,3], time_units=units.s)
+        m.fs.sub = FlowsheetBlock(dynamic=False, time_set = [0, 1], time_units=units.dimensionless)
```

Comment:
Time should not be dimensionless - it does not matter for the test, but we should not accidentally encourage people to do think this is OK.